core/key.go

```diff
@@ -73,14 +74,25 @@ type DataStoreKey struct {
 
 var _ Key = (*DataStoreKey)(nil)
 
+// IndexedField contains information necessary for storing a single
+// value of a field in an index.
+type IndexedField struct {
+	// ID is the id of the field in the schema
+	ID client.FieldID
```

Comment:
todo: It is not super obvious as to why the field ID forms part of the key, can you please document this?

Reply:
I actually realised I didn't need it and planned to get rid of it in a follow up commit.

Reply:
removed